src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ITEMS_NUMBER = 10;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_ITEMS_NUMBER];
+    }
+
     @Override
     public void add(T value) {
-
+        checkToGrow();
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (!indexCheck(index) && index != size) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index
+                    + " is negative or out of bounds with size: " + size);
+        } else if (index == size) {
+            add(value);
+        } else {
```

Comment:
``` suggestion
        checkIndex();
        if (index == size) {
            add(value);
            return;
        }
```

Reply:
not fixed

Reply:
> not fixed

What should i fix? I dont understand

Reply:
Hi! All questions should be discussed in java chat, write there, please
Yoy have one extra IF there